### PR TITLE
[camera] Reland implementation of flip camera while recording. App facing changes

### PR DESCRIPTION
**Pull request by BradenBagby**
This is the final PR for the app facing camera package for the original flip/change camera PR on the old Plugins repo here: https://github.com/flutter/plugins/pull/6478

The application interface was merged here: https://github.com/flutter/plugins/pull/7011
The platform packages were merged here: https://github.com/flutter/packages/pull/3460

### CODE REVIEW

**Review by bparrishMines — APPROVED** [review]
LGTM with a pull from main.

@stuartmorgan or @camsim99 for secondary review

**Comment by BradenBagby** [comment]
Is this test an issue with this PR or is the test broken in general?  flutter pub get in the camera package works fine locally for me

File: packages/camera/camera/example/integration_test/camera_test.dart [hunk 1]
```diff
@@ -265,6 +265,49 @@ void main() {
...
+      cameras[0],
+      ResolutionPreset.low,
+      enableAudio: false,
+    );
+
+    await controller.initialize();
+    await controller.prepareForVideoRecording();
+
+    await controller.startVideoRecording();
+    sleep(const Duration(milliseconds: 500));
```
Comment (stuartmorgan) [inline packages/camera/camera/example/integration_test/camera_test.dart:284]: Carrying comments over from implementation tests: do we really need all these half-second sleeps? If so, why?

File: packages/camera/camera/test/camera_value_test.dart [hunk 2]
```diff
@@ -13,26 +13,28 @@ import 'package:flutter/cupertino.dart';
...
+          flashMode: FlashMode.auto,
+          exposureMode: ExposureMode.auto,
+          exposurePointSupported: true,
+          focusMode: FocusMode.auto,
+          deviceOrientation: DeviceOrientation.portraitUp,
+          lockedCaptureOrientation: DeviceOrientation.portraitUp,
+          recordingOrientation: DeviceOrientation.portraitUp,
+          focusPointSupported: true,
+          previewPauseOrientation: DeviceOrientation.portraitUp,
+          description: FakeController.fakeDescription);
```
Comment (stuartmorgan) [inline packages/camera/camera/test/camera_value_test.dart:37]: Nit: Please add the trailing comma, for consistency with the opposite change below.

**Comment by BradenBagby** [comment]
Im not sure why version going from 0.10.4 to 0.10.5 is disallowed here. I must be missing something simple

**Comment by stuartmorgan** [comment]
You're going from `0.10.3+2` to `0.10.5`.

**Comment by BradenBagby** [comment]
This PR should be good now

**Pull request merged**